File: UVA/10038.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
	int n, nums[3002];

	while(scanf("%d", &n)!=EOF){
		bool jolly = true;
		set <int> s;

		for(int i = 0;i<n;i++)
			scanf("%d",&nums[i]);

		for(int i = 0;i<n-1;i++)
			s.insert(abs(nums[i]-nums[i+1]));

		for(int i = 1;i<n-1;i++)
			if(!s.count(i)){
				jolly = false;
				break;
			}		
		
		if(jolly)
			printf("Jolly\n");
		else
			printf("Not jolly\n");
	}

	return 0;
}

File: UVA/10038.py
while True:
	try:
		x = input()
		aux = x.split()
		c = int(aux[0])
		aux = aux[1:]
		y = set()
		ok = True

		for i in range(len(aux)-1):
			y.add(abs(int(aux[i])-int(aux[i+1])))

		for i in range(1,c):	
			if not i in y:
				ok = False
				break
		if ok:
			print('Jolly')
		else:
			print('Not jolly')

	except:
		break